ksql-parser/src/main/java/io/confluent/ksql/parser/tree/AstVisitor.java

```diff
@@ -310,6 +310,22 @@ protected R visitWindow(Window node, C context) {
     return visitNode(node, context);
   }
 
+  protected R visitWindowExpression(WindowExpression node, C context) {
```

Comment:
nit: `final` and elsewher

Reply:
I would leave the change for ASTVisitor and DefaultASTVisitor for another PR since the other subclasses can be updated accordingly.
But I will update the new classes in this PR with final args and variables as suggested.